uploadservice/src/main/java/net/gotev/uploadservice/UploadService.java

```diff
@@ -483,4 +489,22 @@ protected static UploadStatusDelegate getUploadStatusDelegate(String uploadId) {
 
         return delegate;
     }
+
+    /**
+     * Returns the index of the current upload task
+     *
+     * @return
+     */
+    public int getIndexOfCurrentUploadTask() {
+        return completedTasksMap.size() + 1;
```

Comment:
Nitpick: I think `index` is a misleading word here - this is the number of the task in a queue, e.g. 4 out of 5. An index would imply that you can get the same task by passing `4` as a parameter somewhere, which isn't true since the backing structure is a hash map.